src/goto.jl

```diff
@@ -0,0 +1,274 @@
+  return pathitemsmaps
+end
+
+function _searchtoplevelitems(mod::Module, pathitemsmaps::PathItemsMaps)
+  entrypath, paths = modulefiles(mod)
+  if entrypath !== nothing # when Revise approach succeeds, just find items in those files
+    for p ∈ [entrypath; paths]
+      _searchtoplevelitems(p, pathitemsmaps)
+    end
+  else # if Revise approach fails, fallback to parser-based module walk
+    path = parentfile(mod)
+    text = read(path, String)
+    _searchtoplevelitems(text, path, pathitemsmaps)
+  end
+end
+
+function _searchtoplevelitems(path::String, pathitemsmaps::PathItemsMaps)
+  text = read(path, String)
+  parsed = CSTParser.parse(text, true)
+  items = toplevelitems(parsed, text)
+  pathitemsmap = path => items
+  push!(pathitemsmaps, pathitemsmap)
+end
+
+function _searchtoplevelitems(text::String, path::String, pathitemsmaps::PathItemsMaps)
+  parsed = CSTParser.parse(text, true)
+  items = toplevelitems(parsed, text)
+  pathitemsmap = path => items
+  push!(pathitemsmaps, pathitemsmap)
+
+  # module-walk via toplevel `include` call search
+  for item in items
+    if item isa ToplevelCall
+      expr = item.expr
+      if isinclude(expr)
+        nextfile = expr.args[3].val
+        nextpath = joinpath(dirname(path), nextfile)
+        text = read(nextpath, String)
+        _searchtoplevelitems(text, nextpath, pathitemsmaps)
+      end
+    end
+  end
+end
+
+filtertoplevelitem(word, item::ToplevelItem) = false
+function filtertoplevelitem(word, bind::ToplevelBinding)
+  bind = bind.bind
+  bind === nothing ? false : word == bind.name
+end
+function filtertoplevelitem(word, tupleh::ToplevelTupleH)
+  expr = tupleh.expr
+  for arg in expr.args
+    if str_value(arg) == word
+      return true
+    end
+  end
+  return false
+end
+
+function GotoItem(path, bind::ToplevelBinding)
+  expr = bind.expr
+  text = bind.bind.name
+  if CSTParser.has_sig(expr)
+    sig = CSTParser.get_sig(expr)
+    text = str_value(sig)
+  end
+  line = bind.lines.start - 1
+  secondary = path * ":" * string(line)
+  GotoItem(text, path, line, secondary)
+end
+function GotoItem(path::String, tupleh::ToplevelTupleH)
+  expr = tupleh.expr
+  text = str_value(expr)
+  line = tupleh.lines.start - 1
+  secondary = path * ":" * string(line)
+  GotoItem(text, path, line, secondary)
+end
+
+## regenerate toplevel symbols
+
+handle("regeneratesymbols") do data
+  @destruct [
+    mod || "Main",
+    text || "",
+    path || "untitled"
+  ] = data
+  regeneratesymbols(mod, text, path)
+  nothing
+end
+
+function regeneratesymbols(mod, text, path = "untitled")
+  mod = getmodule(mod)
+
+  if haskey(SYMBOLSCACHE, mod)
+    parsed = CSTParser.parse(text, true)
+    items = toplevelitems(parsed, text)
+    push!(SYMBOLSCACHE[mod], path => items)
+  else
+    # there is no cache
+    SYMBOLSCACHE[mod] = searchtoplevelitems(mod, text, path)
+  end
+end
+
+## TODO: generate toplevel symbols cache for project modules, like `regenerateCache` for docs
```

Comment:
Any idea ? https://julialang.slack.com/archives/C6A044SQH/p1570026949051100

Reply:
Are you looking for `Base.find_package("Atom")`?

Reply:
Why do I forget that.
yes, exactly. thanx.